### PR TITLE
Make `master` a placeholder.

### PR DESCRIPTION
**Pull request by Carreau**
The `master` form already has `master` as a placeholder; this (should)
make full use of this; by replacing empty branches by master

### CODE REVIEW

**Comment by yuvipanda** [comment]
Hmm, I feel like we explicitly should not do this, since that will affect
API semantics for non-ui use cases.

**Comment by yuvipanda** [comment]
I take that back! This is actually totally fine and an improvement! We
should merge and deploy this.

**Comment by Carreau** [comment]
I'll attempt to do that later!

File: binderhub/static/index.js [hunk 1]
```diff
@@ -102,7 +104,7 @@ function updateUrl() {
   repo = repo.replace(/^(https?:\/\/)?github.com\//, '');
   // trim trailing or leading '/' on repo
   repo = repo.replace(/(^\/)|(\/?$)/g, '');
-  var ref = $('#ref').val().trim();
+  var ref = $('#ref').val().trim() || 'master';
```
Comment (minrk) [inline binderhub/static/index.js]: empty strings aren't falsy in javascript, so `|| master` will never run. We'd have to add

```javascript
if (ref.length === 0) ref = 'master';
```

to get this behavior.

Comment (Carreau) [reply binderhub/static/index.js]: `||` seem to work for me.... but happy to change. 
screen shot 2017-10-20 at 11 20 56

Comment (Carreau) [reply binderhub/static/index.js]: [According to MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators), empty-string is casted to false.

Comment (minrk) [reply binderhub/static/index.js]: Whoa, not sure where I'm getting my info. Sorry for misleading!

**Pull request merged**